### PR TITLE
Add kopia server connect helper methods

### PR DESCRIPTION
**Pull request by PrasadG193**
Signed-off-by: Prasad Ghangal <prasad.ghangal@gmail.com>

## Change Overview

New pkg kopia with kopia server connect helper methods 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

Will wait for server side implementation for the e2e testing

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by pavannd1** [comment]
Looks good overall. We may have to see if the changes to repository in the latest kopia master would require changes to this. We can merge and follow up.

File: pkg/kopia/connect.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
+package kopia
```
Comment (PrasadG193) [inline pkg/kopia/connect.go:1]: @PrasadG193 add license headers

**Review by pavannd1 — APPROVED** [review]

**Comment by pavannd1** [comment]
@PrasadG193 let's queue this behind #869 if possible. Some API changes would be required when we update Kopia

**Comment by pavannd1** [comment]
@PrasadG193 The other PR merged. Please update the helpers accordingly

File: pkg/kopia/connect.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	opts := &repo.ConnectOptions{
+		PersistCredentials: defaultConnectPersistCredentials,
+		CachingOptions: content.CachingOptions{
+			CacheDirectory:            defaultCacheDirectory,
+			MaxCacheSizeBytes:         int64(defaultDataStoreGeneralContentCacheSizeMB << 20),
+			MaxMetadataCacheSizeBytes: int64(defaultDataStoreGeneralMetadataCacheSizeMB << 20),
+			MaxListCacheDurationSec:   int(defaultConnectMaxListCacheDuration.Seconds()),
+		},
+		HostnameOverride: hostname,
+		UsernameOverride: username,
```
Comment (pavannd1) [inline pkg/kopia/connect.go:73]:
```suggestion
		ClientOptions: repo.ClientOptions{
			Hostname: hostname,
			Username: username,
		},
```

**Pull request merged**